### DISCUSSION TITLE
Multi-animal inference produces >1000 tracks

### DISCUSSION THREAD

**Discussion by rentazillla** [talmolab/sleap#1416] [Help!]
Hello all,

I am trying to build a multi-animal top-down pipeline for tracking mouse behavior. I am fairly happy with the instance predictions but when I go to run the inference for producing track assignments, I get 1077 tracks which is far too many to delete one by one. Any ideas on what I am doing wrong or what I should try next?

Here are some screenshots showing metrics from the trained model and hyperparameters:
![Screenshot 2023-07-26 115722](https://github.com/talmolab/sleap/assets/14893909/2a256d33-f583-445b-ba40-f7b7a35791a4)

![Screenshot 2023-07-26 115757](https://github.com/talmolab/sleap/assets/14893909/2557ce16-ecaa-4adb-9417-40b0056229fd)

![Screenshot 2023-07-26 115803](https://github.com/talmolab/sleap/assets/14893909/d74090e4-7787-42aa-93a7-7b87fe2afc1b)

![Screenshot 2023-07-26 115809](https://github.com/talmolab/sleap/assets/14893909/a61f380d-2a66-425f-b79b-ac5681c0491e)

![Screenshot 2023-07-26 115517](https://github.com/talmolab/sleap/assets/14893909/333bfe6c-8a93-43f8-b823-ffe48c08af65)

Thanks for any help or advice you might be willing to share!

**Comment by roomrys** [comment]
Hi @rentazillla,

The first thing I would recommend is to consider using retracking which should speed-up the tracking iterations quite a bit (by bypassing inference). SLEAP v1.3.0-current (1.3.1) allows retracking on predicted instances only, so you will need to **save a new version** of your project and Labels > Custom Instance Delete all user instances across all videos. From there, you can set the Training/Inference Pipeline Type to None to bypass inference.

To start, I would try a simple matching method first with instance matching then with centroid matching. I would also try deselecting the "Use max (non-robust)".

Let us know how it goes.

Thanks,
Liezl

**Reply by rentazillla** [reply]
Thanks very much @roomrys 

I saved a new project and deleted the user instances across all the videos and followed your suggestions regarding the settings. However, I am still getting >1000 tracks. For the training/inference pipeline, do I need to delete all the previous predicted instances and "Run Training..." before performing "Run Inference..." ? I tried that and got an error.

If I keep all the predicted frames (no user frames) from the previous version and "Run Inference..." with the Pipeline type set to None I do get track predictions but I still get >1000 tracks. I tried doing the same "Run Inference..." procedure a second time and did not get any improvement. If anything it added additional tracks

**Reply by roomrys** [reply]
Hi @rentazillla,

No, do not delete the previous predicted instances. We are bypassing the inference step so no new predictions will be output. This is to save us time in finding the correct tracking parameters. That being said, we will also not be training a new model, so feel free to jump to "Run Inference..." which with the Inference Pipeline Type set to none will just perform (re)tracking.

I also noticed that in your initial screenshot, the "Predict On" field was set to the a single frame only. I want to double check that this is always set to "entire current video" or "all videos" as tracking requires sequential frames.

Thanks,
Liezl

**Reply by rentazillla** [reply]
I did try the above suggestions again today (tried yesterday too) and I'm still getting >600 tracks labeled. I am using "Predict on" for "entire current video." I tried simple tracking with instance matching first, then saved and repeated with centroid tracking and no luck thus far.

**Comment by talmo** [comment]
Side note: Even if we have switches, it's often desirable to have a global max number of tracks. Relevant issue: #1308
